### PR TITLE
[App Service] BREAKING CHANGE: Redact tokens output on deployment source update-token

### PR DESCRIPTION
**Pull request by kamperiadis**
**Related command**
`az functionapp deployment source update-token`
`az webapp deployment source update-token`

**Description**
Currently, these two commands return the tokens as part of the output. In this PR, we are redacting the tokens so prevent accidental security leaks through CI logs.

**Testing Guide**
1. Create functionapp (`az functionapp create`)
2. Update token (`az functionapp deployment source update-token`)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2052,12 +2052,17 @@ def config_source_control(cmd, resource_group_name, name, repo_url, repository_t
 def update_git_token(cmd, git_token=None):
     '''
     Update source control token cached in Azure app service. If no token is provided,
-    the command will clean up existing token.
+    the command will clean up existing token. Note that tokens are now redacted in the result.
```
Comment (kamperiadis) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2055]: Question for reviewer: I do not see any help.py file with this text. Does that mean that this is the text used for the documentation [here](https://learn.microsoft.com/en-us/cli/azure/functionapp/deployment/source?view=azure-cli-latest#az-functionapp-deployment-source-update-token)?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -2052,12 +2052,17 @@ def config_source_control(cmd, resource_group_name, name, repo_url, repository_t
...
     '''
     client = web_client_factory(cmd.cli_ctx)
     from azure.mgmt.web.models import SourceControl
     sc = SourceControl(name='not-really-needed', source_control_name='GitHub', token=git_token or '')
-    return client.update_source_control('GitHub', sc)
+    response = client.update_source_control('GitHub', sc)
+    logger.warning('Tokens have been redacted.')
```
Comment (kamperiadis) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2061]: There is no command we provide that could provide these tokens

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2055]: yes, correct

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2061]: Does this mean it is hard to rerun test in live mode to validate this change?

Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2061]: I just added a live test.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**